support/refc/_datatypes.h

```diff
@@ -50,15 +45,47 @@ typedef struct {
   // followed by type-specific payload.
 } Value;
 
-typedef struct {
-  Value_header header;
-  uint8_t ui8;
-} Value_Bits8;
+/* expcted at least 4bytes for Value_header alignment. */
+#define idris2_vp_is_unboxed(p) ((uintptr_t)(p)&3)
 
-typedef struct {
-  Value_header header;
-  uint16_t ui16;
-} Value_Bits16;
+#define idris2_vp_int_shift                                                    \
+  ((sizeof(uintptr_t) >= 8 && sizeof(Value *) >= 8) ? 32 : 16)
+
+#define idris2_vp_to_Bits64(p) (((Value_Bits64 *)(p))->ui64)
+
+#if !defined(UINTPTR_WIDTH)
+#define idris2_vp_to_Bits32(p)                                                 \
+  ((idris2_vp_int_shift == 16)                                                 \
+       ? (((Value_Bits32 *)(p))->ui32)                                         \
+       : ((uint32_t)((uintptr_t)(p) >> idris2_vp_int_shift)))
+#define idris2_vp_to_Int32(p)                                                  \
+  ((idris2_vp_int_shift == 16)                                                 \
+       ? (((Value_Int32 *)(p))->i32)                                           \
+       : ((int32_t)((uintptr_t)(p) >> idris2_vp_int_shift)))
+
+#elif UINTPTR_WIDTH >= 64
+#define idris2_vp_to_Bits32(p)                                                 \
+  ((uint32_t)((uintptr_t)(p) >> idris2_vp_int_shift))
+#define idris2_vp_to_Int32(p): ((int32_t)((uintptr_t)(p) >> idris2_vp_int_shift))
+
+#elif UINTPTR_WIDTH >= 32
+#define idris2_vp_to_Bits32(p) (((Value_Bits32 *)(p))->ui32)
+#define idris2_vp_to_Int32(p) (((Value_Int32 *)(p))->i32)
+
+#else
+#error "unsupported uintptr_t width"
```

Comment:
I think, the idea that the target architecture of the `refc` backend is assumed to have pointer type to have at least 32 bits should be documented, at least in the changelog, because this is potentially a breaking thing since Idris has some successful stories of running on 16-bit architectures in the past, so this information should be present somewhere.